File: app/models/category.py
"""Category model."""

from sqlalchemy import Column, DateTime, Integer, String, Text
from sqlalchemy.orm import relationship

from app.core.datetime import utc_now
from app.db.database import Base


class Category(Base):
    """Category table used to group articles."""

    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), unique=True, index=True, nullable=False)
    description = Column(Text, nullable=True)
    created_at = Column(DateTime, default=utc_now, nullable=False)
    updated_at = Column(
        DateTime,
        default=utc_now,
        onupdate=utc_now,
        nullable=False,
    )

    articles = relationship("Article", back_populates="category")

File: frontend/src/types/category.ts
export type CategorySimple = {
  id: number;
  name: string;
};

export type CategoryRead = {
  id: number;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
};

export type CategoryCreateRequest = {
  name: string;
  description?: string | null;
};
